File: src/main/kotlin/Java/RPG/Guerrero.java
package Java.RPG;

public class Guerrero extends Personaje {

    public Guerrero(String nombre, int puntosDeVida, int ataque) {
        super(nombre, puntosDeVida, ataque);
    }

    @Override
    public void atacar(Personaje objetivo) {
        System.out.println(nombre + " ataca a " + objetivo.getNombre() + " con su espada.");
        objetivo.setPuntosDeVida(objetivo.getPuntosDeVida() - ataque);
        System.out.println(objetivo.getNombre() + " tiene ahora " + objetivo.getPuntosDeVida() + " puntos de vida.");
    }
}

File: src/main/kotlin/Kotlin/RPG/Guerrero.kt
package Kotlin.RPG

class Guerrero(nombre: String, puntosDeVida: Int, ataque: Int) : Personaje(nombre, puntosDeVida, ataque) {
    override fun atacar(objetivo: Personaje) {
        println("$nombre ataca a ${objetivo.nombre} con su espada.")
        objetivo.puntosDeVida -= ataque
        println("${objetivo.nombre} tiene ahora ${objetivo.puntosDeVida} puntos de vida.")
    }
}
